### PR TITLE
Dynamic meeting form

### PR DESCRIPTION
**Pull request by chriswedgwood**
These are changes made after form was reviewed in staging.

### CODE REVIEW

File: aalondon/static/js/components/MeetingForm.js [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+          </div>
+          <div className="form-group">
+            <label htmlFor="notes">Notes</label>
+            <Field name="notes" component="textarea" type="text" className={'form-control' + (errors.notes && touched.notes ? ' is-invalid' : '')} />
+            <ErrorMessage name="notes" component="div" className="invalid-feedback" />
+          </div>
+
+          <div className="auto-grid" role="group" aria-labelledby="checkbox-group">
+          {formType !== 'online' &&
+            <><div><span className="checkbox-title" htmlFor="wheelchair">Wheel Chair</span>
```
Comment (tintinnabulate) [inline aalondon/static/js/components/MeetingForm.js:367]: `<span ...>Wheelchair-accessible</span>`

File: aalondon/static/js/components/MeetingForm.js [hunk 2]
```diff
@@ -0,0 +1,445 @@
...
+            <div>
+              <span className="checkbox-title" htmlFor="creche">Creche</span>
+              <SemanticField
+                name="creche"
+                component={Checkbox}
+              />
+
+            </div></>}
+            <div>
+              <span className="checkbox-title" htmlFor="signed">Signed</span>
```
Comment (tintinnabulate) [inline aalondon/static/js/components/MeetingForm.js:382]: `<span ...>Sign-language interpreted</span>`

File: aalondon/static/js/components/MeetingForm.js [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+              id={"day"}
+              value={[]}
+              className={'form-control' + (errors.day && touched.day ? ' is-invalid' : '')}
+
+            />
+
+            <ErrorMessage name="day" component="div" className="invalid-feedback" />
+          </div>
+          <div className="form-group">
+            <label htmlFor="submission">Submission</label>
```
Comment (tintinnabulate) [inline aalondon/static/js/components/MeetingForm.js:282]: `<label ...>Is this a new entry to AALondon, or an update to an existing entry?</label>`

File: aalondon/static/js/components/MeetingForm.js [hunk 4]
```diff
@@ -0,0 +1,445 @@
...
+    text: 'London Croydon Epsom & Sutton',
+    value: 'London Croydon Epsom & Sutton',
+  },
+  {
+    key: 'London North Kent',
+    text: 'London North Kent',
+    value: 'London North Kent',
+  },
+  {
+    key: 'London South East (East)"',
```
Comment (tintinnabulate) [inline aalondon/static/js/components/MeetingForm.js:93]: Remove the `"` from the end of these (`key`, `text` and `value`).

**Review by tintinnabulate — COMMENTED** [review]
It's awesome that this has been done. I took at look at staging and had a few review comments. Nothing major.

File: aalondon/static/js/MeetingFormSelector.js [hunk 5]
```diff
@@ -31,21 +29,12 @@ const formTypes = [
...
-    else if (formType == 'hybrid') {
-        activeForm = <HybridForm formType={formType}/>
-
-    }
+  
+    activeForm = formType === 'select' ? <></> : <MeetingForm formType={formType}/> 
+    
     return (
         <>
             <label htmlFor="notes">Meeting Type(Hybrid,Online or Physical)</label>
```
Comment (tintinnabulate) [inline aalondon/static/js/MeetingFormSelector.js:40]: `<label ...>Meeting Type (Online only, Face to Face, or Hybrid)</label>`

File: aalondon/static/js/components/MeetingForm.js [hunk 6]
```diff
@@ -0,0 +1,445 @@
...
+        childFriendly: false,
+        outdoors: false,
+        creche: false,
+        temporaryClosure: false
+      }}
+      validationSchema={validationSchema}
+      onSubmit={fields => {
+
+
+        alert('SUCCESS!! :-)\n\n' + JSON.stringify(fields, null, 4))
```
Comment (tintinnabulate) [inline aalondon/static/js/components/MeetingForm.js:257]: Hopefully you know to remove this alert when this is merged to production.

**Review by tintinnabulate — COMMENTED** [review]
A few more review comments (See in-line with the code).

Also, we need the following two additional fields:

1. **Group Name**. Required field. Simple text field. I think this should be the first field in the form, after "Meeting Type" has been selected.
2. **GDPR Acceptance** Required field. Boolean tickbox. This can be the very last field in the form.

**Comment by chriswedgwood** [comment]
👍

The alert is just a placeholder for now until you develop the post endpoint

**Pull request merged**